main.py

```diff
@@ -1,37 +1,99 @@
+from copy import deepcopy
+
 MIN_NUMBER = -2147483648
 MAX_NUMBER = 2147483647
+options = {
+    1: "Find the determinant of the matrix"
+}
+GET_DETERMINANT = 1
 
 
 def get_input(msg, quantity, interval):
-    condition = False
-    while not condition:
+    # Get input from user: msg - message to show, quantity of the accepted parameters that are included in the interval
+    is_good_ans = False
+    while not is_good_ans:
         ans = list(input(msg).split())
-        condition = (len(ans) == quantity or quantity == -1)
+        is_good_ans = (len(ans) == quantity or quantity == -1)  # -1 means unlimited quantity of the parameters
         try:
             for element in ans:
                 if not (int(element) in interval):
-                    condition = False
+                    is_good_ans = False
                     break
         except:
-            condition = False
-        if condition:
+            is_good_ans = False
+        if is_good_ans:
             if quantity == 1:
                 ans = int(ans[0])
             return ans
         else:
-            print("Something went wrong...")
+            print("Something went wrong...")  # TODO: More detailed exceptions messages
+
+
+def get_option():  # Print all the options and get user's response
+    print("Options: ")
+    for i in range(len(options)):
+        print(str(i + 1) + ': ' + options[i + 1] + ';')
+    print()
+    return get_input("Enter: ", 1, range(1, len(options) + 1))
+
+
+class Matrix:
+    def __init__(self):
+        # Create the Matrix size n*m:
+        self.matrix = []
+        self.n = int()
+        self.m = int()
+        # TODO: Minor: get_height/get_weight
 
+    def init_matrix(self, lines):  # Method: Initialization the matrix from the keyboard
```

Comment:
The idea was that your Matrix class does NOT know anything about both console and keyboard. Imagine you want to reuse this class in a web-application or mobile. Then you will not be able to do that becuase of console and keyboard dependency. The idea to have this method and get_matrix() outside the Matrix class.

Ideally there must be an outside method like 'read_matrix' which reads it from keyboard and returns already built Matrix object